### ISSUE TITLE
Regression `builtins.getContext` / `outputs = [ "" ]`

### ISSUE THREAD

**Issue by roberth** [NixOS/nix#7655]
**Describe the bug**

`builtins.getContext` always returns `{ outputs = [ "" ]; }` for build outputs.

**Steps To Reproduce**

`cd nixpkgs; nix repl .`

Expected:

Actual:

```
nix-repl> :p builtins.getContext (writeText "hi" "hi").outPath
{ "/nix/store/axry75xwlchnnrqp263d05yhr89cv3d0-hi.drv" = { outputs = [ "" ]; }; }

nix-repl> :p builtins.getContext nix.doc.outPath
{ "/nix/store/56n2kf5pha0m8cs19vgzhv5j0d7mr96n-nix-2.12.0.drv" = { outputs = [ "" ]; }; }

```

**Expected behavior**

```
nix-repl> :p builtins.getContext (writeText "hi" "hi").outPath
{ "/nix/store/axry75xwlchnnrqp263d05yhr89cv3d0-hi.drv" = { outputs = [ "out" ]; }; }

nix-repl> :p builtins.getContext nix.doc.outPath                    
{ "/nix/store/56n2kf5pha0m8cs19vgzhv5j0d7mr96n-nix-2.12.0.drv" = { outputs = [ "doc" ]; }; }

```

**`nix-env --version` output**

master, 2.13.1

**Additional context**

 - Found in https://github.com/NixOS/nixpkgs/pull/211290#issuecomment-1399282084

**Priorities**

Add :+1: to [issues you find important](https://github.com/NixOS/nix/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

**Comment by Ericson2314** [comment]
Have fix, just writing tests.

**Comment by Ericson2314** [comment]
`tests/lang/eval-okay-context-introspection.nix` is already a test, I am confused why it is passing.

**Comment by Ericson2314** [comment]
Oh nevermind.

**Comment by roberth** [comment]
The fix has been released in 2.13.2.
